src/acl_auto_configure.cpp

```diff
@@ -493,6 +493,61 @@ bool acl_load_device_def_from_str(const std::string &config_str,
                                 counters);
   }
 
+  // Read device global information.
+  unsigned int num_device_global = 0;
+  if (result && counters.back() > 0) {
+    result =
+        read_uint_counters(config_str, curr_pos, num_device_global, counters);
+    devdef.num_device_global = num_device_global;
+
+    // read total number of fields in device global
+    int total_fields_device_global = 0;
+    if (result) {
+      result = read_int_counters(config_str, curr_pos,
+                                 total_fields_device_global, counters);
+    }
+
+    for (auto i = 0U; result && (i < num_device_global);
+         i++) { // device_global_memories
+      counters.emplace_back(total_fields_device_global);
+
+      // read device global name
+      std::string device_global_name;
+      if (result && counters.back() > 0) {
+        result = read_string_counters(config_str, curr_pos, device_global_name,
+                                      counters);
+      }
```

Comment:
What happens if `result` evaluates to `false` and `device_global_name` is uninitialized, i.e. the empty string?

Reply:
if any one of the result is false, it will get propagated until the end of the function (i.e once its false, it will be false for the lifetime of the function call). and all the acceleration info is empty then output error.